containers/base_golang.go

```diff
@@ -7,11 +7,8 @@ import (
 )
 
 // GolangContainer returns a dagger container with everything set up that is needed to build Grafana's Go backend or run the Golang tests.
-func GolangContainer(d *dagger.Client, platform dagger.Platform, base string) *dagger.Container {
+func GolangContainer(d *dagger.Client, base string) *dagger.Container {
 	opts := dagger.ContainerOpts{}
-	if platform != "" {
-		opts.Platform = platform
-	}
 
 	container := d.Container(opts).From(base)
```

Comment:
So this means that on darwin/arm64 the container will also have that architecture. Is that intended?

Reply:
after some testing there's definitely bugs in here. I'll create an issue and re-evaluate how we're going to approach this.